### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'encode'

### ISSUE THREAD

**Issue by CrimsonGlory** [erocarrera/pefile#184]
```
import pefile
file = open('c14f6249cd75fff28cd04371f28da4214600b105.codex','r').read()
library = pefile.PE(data=file, fast_load=True)
dir_folders = [
                    pefile.DIRECTORY_ENTRY.get('IMAGE_DIRECTORY_ENTRY_IMPORT'),
                    pefile.DIRECTORY_ENTRY.get('IMAGE_DIRECTORY_ENTRY_EXPORT'),
                    pefile.DIRECTORY_ENTRY.get('IMAGE_DIRECTORY_ENTRY_TLS'),
                    pefile.DIRECTORY_ENTRY.get('IMAGE_DIRECTORY_ENTRY_SECURITY'),
                    pefile.DIRECTORY_ENTRY.get('IMAGE_DIRECTORY_ENTRY_RESOURCE')]
dir_folders
[1, 0, 9, 4, 2]
library.parse_data_directories(directories=dir_folders)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/pefile.py", line 2370, in parse_data_directories
    value = entry[1](dir_entry.VirtualAddress, dir_entry.Size)
  File "/usr/local/lib/python2.7/dist-packages/pefile.py", line 2979, in parse_resources_directory
    self.parse_version_information(rt_version_struct)
  File "/usr/local/lib/python2.7/dist-packages/pefile.py", line 3112, in parse_version_information
    versioninfo_string = versioninfo_string.encode('utf-8', 'backslashreplace')
AttributeError: 'NoneType' object has no attribute 'encode'
```

File:
```
http://www.mediafire.com/file/7ld7t54u3w7fdz3/c14f6249cd75fff28cd04371f28da4214600b105.zip
```
Password: ```codex```
(Do not execute it. it is malware)

```
$ pip list | grep pefile
pefile (2016.3.28)
$ python --version
Python 2.7.12
```

**Comment by CrimsonGlory** [comment]
```
>>> library.parse_data_directories(directories=[1, 0, 9, 4])
>>> library.parse_data_directories(directories=[2])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/pefile.py", line 2370, in parse_data_directories
    value = entry[1](dir_entry.VirtualAddress, dir_entry.Size)
  File "/usr/local/lib/python2.7/dist-packages/pefile.py", line 2979, in parse_resources_directory
    self.parse_version_information(rt_version_struct)
  File "/usr/local/lib/python2.7/dist-packages/pefile.py", line 3112, in parse_version_information
    versioninfo_string = versioninfo_string.encode('utf-8', 'backslashreplace')
```

**Comment by CrimsonGlory** [comment]
(I confirm this is fixed on pefile 2017.11.5. thanks)
